snap/info.go

```diff
@@ -157,11 +163,21 @@ func (s *Info) DataDir() string {
 	return filepath.Join(dirs.SnapDataDir, s.Name(), s.strRevno())
 }
 
+// CommonDataDir returns the data directory common between versions of the snap.
+func (s *Info) CommonDataDir() string {
+	return filepath.Join(dirs.SnapDataDir, s.Name(), "common")
+}
+
 // DataHomeDir returns the per user data directory of the snap.
 func (s *Info) DataHomeDir() string {
 	return filepath.Join(dirs.SnapDataHomeGlob, s.Name(), s.strRevno())
 }
 
+// CommonDataHomeDir returns the per user data directory common between versions of the snap.
+func (s *Info) CommonDataHomeDir() string {
+	return filepath.Join(dirs.SnapDataHomeGlob, s.Name(), "common")
```

Comment:
DataHomeDir is my fault refactoring away and not noticing that it involved actually a glob, not sure one method and one dirs pkg variable can cover all our use cases though, it seems we need to produce:
<globRootDir>/home/*/snap/...   "/root/snap/..."  and "$HOME/snap/..."